service/src/main/java/edu/tamu/catalog/service/FolioCatalogService.java

```diff
@@ -367,6 +399,78 @@ private String getServicePointDisplayName(String servicePointId) {
         return  null;
     }
 
+    /**
+     * Get FOLIO User by the user's UIN via OKAPI.
+     *
+     * @param uin The user's UIN.
+     * @return request type
+     */
+    private JsonNode getUserByUin(String uin) {
+        if (Objects.isNull(uin)) {
+            return null;
+        }
+
+        String path = "%s/bl-users?query=(externalSystemId==\"{uin}\")&limit=2";
+        String url = String.format(path, properties.getBaseOkapiUrl(), uin);
+        String message = "user using external system id";
+
+        logger.debug("Asking for User from: {}", url);
+
+        JsonNode response = okapiRequestJsonNode(url, HttpMethod.GET, message, uin);
+
+        if (Objects.nonNull(response) && response.isObject()) {
+            JsonNode users = response.at("/compositeUsers");
+
+            if (Objects.nonNull(users) && users.isArray()) {
+                if (Objects.nonNull(users.get(1))) {
+                    throw new HttpServerErrorException(HttpStatus.INTERNAL_SERVER_ERROR,
+                        "Found multiple users with the same external system id");
+                }
+
+                JsonNode user = users.get(0);
+                if (Objects.nonNull(user) && user.isObject()) {
+                    return user.at("/users");
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Get FOLIO automated block status via OKAPI.
+     *
+     * @param userId The user's UIN.
+     * @return automated block status
+     */
+    private Boolean getAutomatedBlockStatus(String userId) {
+        if (Objects.isNull(userId)) {
+            return null;
```

Comment:
These type of auth guards only seem elegant when the calling function is called without the ability to perform check on the argument passed in. Otherwise, I feel this promotes a pattern of null checking every argument in every method.